opentasks/src/test/java/org/dmfs/tasks/notification/signals/ToggledTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.tasks.notification.signals;
+
+import android.app.Notification;
+
+import org.junit.Test;
+
+import static org.dmfs.tasks.utils.BitFlagUtils.containsFlag;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+
+/**
+ * Unit test for {@link Toggled}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class ToggledTest
+{
+    @Test
+    public void testValidFlags()
+    {
+        new Toggled(Notification.DEFAULT_SOUND, true, new NoSignal());
+        new Toggled(Notification.DEFAULT_VIBRATE, true, new NoSignal());
+        new Toggled(Notification.DEFAULT_LIGHTS, true, new NoSignal());
+    }
+
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testInValidFlag()
+    {
+        new Toggled(15, true, new NoSignal());
```

Comment:
Same here. If I never call `value()` why shouldn't I create a `Toggled` object with invalid params? Who cares if I don't use the value it returns?
A test case should test actual usage. But this test doesn't use the object it just creates one.

It's like building a car with just three wheels. That's totally fine if you don't intend to drive it, right?

This probably boils down to the question if a constructor should be code free (including validation) or not and whether eager validation is better than lazy validation. I think the latter one is more "declarative". I mean I can declare a trunk to be a car and that's fine if I don't intend to start the engine and actually drive with it (that's why it works pretty well for kids). If I try to start the engine of a trunk I'll fail badly, but not before.
In our code we don't care if the car is actually a trunk as long as we can start the engine. That's why we have test cases.
Or to go with the three wheel car example, we don't care how many wheels it has if it passes the "drive" test. Or do we?

Reply:
Moved this validation from ctor to method. Not sure we even need it, but it doesn't hurt. Updated the tests accordingly to call on `value()`.
(Personally, I don't have a clear opinion on the eager or lazy validation, in general I think fail-fast is good, but this declarative oop approach doesn't follow that, so not sure what fits best.)